### ISSUE TITLE
composer self-update fails after upgrading to openssl 1.1.1

### ISSUE THREAD

**Issue by swiffer** [oerdnj/deb.sury.org#990]
after upgrading to openssl (1.1.1-2+ubuntu18.04.1+deb.sury.org+1) from the PHP PPA i get the following error when running `composer selfupdate`

```bash
  [RuntimeException]
  SHA384 is not supported by your openssl extension, could not verify the phar file integrity
```

**Comment by oerdnj** [comment]
What PHP version? I certain that SHA384 is still in OpenSSL.

**Comment by swiffer** [comment]
7.2.11-4+ubuntu18.04.1+deb.sury.org+1

**Comment by oerdnj** [comment]
Ok, so compare these two lists:
OpenSSL 1.1.0
```
# php7.2 -r 'print_r(openssl_get_md_methods());'
Array
(
    [0] => BLAKE2b512
    [1] => BLAKE2s256
    [2] => MD4
    [3] => MD5
    [4] => MD5-SHA1
    [5] => RIPEMD160
    [6] => SHA1
    [7] => SHA224
    [8] => SHA256
    [9] => SHA384
    [10] => SHA512
    [11] => blake2b512
    [12] => blake2s256
    [13] => md4
    [14] => md5
    [15] => md5-sha1
    [16] => ripemd160
    [17] => sha1
    [18] => sha224
    [19] => sha256
    [20] => sha384
    [21] => sha512
    [22] => whirlpool
)
```

OpenSSL 1.1:
```
# php7.2 -r 'print_r(openssl_get_md_methods());'
Array
(
    [0] => blake2b512
    [1] => blake2s256
    [2] => md4
    [3] => md5
    [4] => md5-sha1
    [5] => ripemd160
    [6] => sha1
    [7] => sha224
    [8] => sha256
    [9] => sha3-224
    [10] => sha3-256
    [11] => sha3-384
    [12] => sha3-512
    [13] => sha384
    [14] => sha512
    [15] => sha512-224
    [16] => sha512-256
    [17] => shake128
    [18] => shake256
    [19] => sm3
    [20] => whirlpool
)
```

Looks like OpenSSL 1.1.1 only provides the lowercase version of algorithm names and composer is relying on the uppercase variants.

I am not sure who to blame here :-)

**Comment by oerdnj** [comment]
Opened issue with composer here: https://github.com/composer/composer/issues/7769

**Comment by swiffer** [comment]
thanks - answered already over there - seems to be solved in 1.7.3
